src/toucan/util.clj

```diff
@@ -9,3 +10,11 @@
   [k]
   (when k
     (s/replace (str k) #"^:" "")))
+
+(defn lower-case
+  "Locale-agnostic version of `clojure.string/lower-case`.
+  `clojure.string/lower-case` uses the default locale in conversions, turning
+  `ID` into `ıd`, in the Turkish locale. This function always uses the
+  `Locale/US` locale."
+  [s]
```

Comment:
Can you tag `s` like `[^CharSequence s]`? Otherwise the Clojure runtime has to do expensive reflection to find the corresponding `toString` method at runtime. If you do `(set! *warn-on-reflection* true)` when you're developing in the REPL it will warn you about stuff like this

Reply:
Would it not then be more prudent to tag it as a `String` and remove the `toString`, as done [here](https://github.com/clojure/java.jdbc/commit/1f1ff8144b4a9a4447b2a2fbb9e592f90ccd2659#diff-3a9a3475f16407a50d922fad53957b01R542)?

Reply:
I looked at the source for `clojure.string/lower-case` and I think what they do makes more sense:

```clj
(defn ^String lower-case
  "Converts string to all lower-case."
  {:added "1.2"}
  [^CharSequence s]
  (.. s toString toLowerCase))
```

`CharSequence` is an interface that includes the `toString` method and `String` is one of several classes that implement it. It doesn't really matter either way tho just as long as it doesn't have reflection warnings and fixes the issue. I don't think we use anything besides regular strings in Toucan

Reply:
Fixed :wrench: